### PR TITLE
feat(storefront): BCTHEME-293 Product list item Quick view button is not tabbable

### PR DESCRIPTION
**Pull request by yurytut1993**
#### What?

This PR has update for quick view button of product list item. Now this button is accessible through keyboard  

#### Tickets / Documentation

[Ticket](https://jira.bigcommerce.com/browse/BCTHEME-293)

#### Screenshots (if appropriate)

When focus in product list item - quick view button in it is visible and accessible via focus
Screenshot 2020-11-11 at 15 37 07
Screenshot 2020-11-11 at 15 37 15

### CODE REVIEW

File: assets/scss/layouts/products/_productList.scss [hunk 1]
```diff
@@ -23,6 +23,21 @@
...
+        }
+    }
+
+    &.focus-within {
+        .listItem-figureBody {
+            @include breakpoint("large") {
+                opacity: 1;
+            }
+        }
+    }
```
Comment (BC-tymurbiedukhin) [inline assets/scss/layouts/products/_productList.scss:40]: Could we change it to `&:focus-within, &.focus-within {`?

File: templates/components/products/list-item.html [hunk 2]
```diff
@@ -19,18 +26,19 @@
...
             fallback_size=theme_settings.productgallery_size
             lazyload=theme_settings.lazyload_mode
             default_image=theme_settings.default_image_product
         }}
+        </a>
         {{#unless hide_product_quick_view}}
             {{#if theme_settings.show_product_quick_view}}
                 <div class="listItem-figureBody">
-                    <a {{#if settings.data_tag_enabled}} data-event-type="product-click" {{/if}} class="button button--small listItem-button quickview" data-product-id="{{id}}">{{lang 'products.quick_view'}}</a>
+                    <button {{#if settings.data_tag_enabled}} data-event-type="product-click" {{/if}} class="button button--small listItem-button quickview" data-product-id="{{id}}">{{lang 'products.quick_view'}}</button>
```
Comment (BC-tymurbiedukhin) [inline templates/components/products/list-item.html:41]: Could we add `type='button'` here?

File: templates/components/products/list-item.html [hunk 3]
```diff
@@ -4,6 +4,13 @@
     <article class="listItem">
 {{/if}}
     <figure class="listItem-figure">
+        <a href="{{url}}"
+           class="listItem-figure__link"
+           aria-label="{{name}},{{> components/products/product-aria-label}}"
+           {{#if settings.data_tag_enabled}}
```
Comment (BC-tymurbiedukhin) [inline templates/components/products/list-item.html:10]: What is this setting responsible for?

Comment (yurytut1993) [reply templates/components/products/list-item.html:10]: For image to be a link to a product (see first screen)

File: templates/components/products/list-item.html (hunk 2 above)
Comment (yurytut1993) [reply templates/components/products/list-item.html:41]: yes

File: assets/scss/layouts/products/_productList.scss (hunk 1 above)
Comment (yurytut1993) [reply assets/scss/layouts/products/_productList.scss:40]: yes

Comment (BC-tymurbiedukhin) [reply assets/scss/layouts/products/_productList.scss:40]: Suggested approach does not work in Edge and IE

File: templates/components/products/list-item.html (hunk 3 above)
Comment (BC-tymurbiedukhin) [reply templates/components/products/list-item.html:10]: What is the diff between existing of setting or not?

Comment (yurytut1993) [reply templates/components/products/list-item.html:10]: Without this link user will not have possibility to click on image to be redirected on PDP

**Comment by BC-tymurbiedukhin** [comment]
LGTM

**Review by BC-tymurbiedukhin — APPROVED** [review]

**Review by bc-alexsaiannyi — APPROVED** [review]

**Pull request merged**